pkg/kubectl/cmd/apply.go

```diff
@@ -193,6 +196,14 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 		return err
 	}
 
+	var openapiSchema openapi.Resources
+	if cmdutil.GetFlagBool(cmd, "openapi-patch") {
+		openapiSchema, err = f.OpenAPISchema()
```

Comment:
this introduces a failure mode that does not fall back to struct info gracefully

Reply:
Fixed. Thanks for catching this.

Reply:
> Fixed. Thanks for catching this.

this means there was no test for this case, right? should add one to ensure we don't regress.

Reply:
I will add a test for it.

Reply:
Done. Add test cases that return error when fetching openapi spec.